src/api.yaml

```diff
@@ -0,0 +1,1285 @@
+                    
+                    
+                    If any of the files are already associated with another message then the
+                    association is ignored.
+                    
+                    
+                    When submitting a message *edit* this field must contain the IDs of any newly
+                    uploaded files that are part of the edit, existing attachment IDs may be
+                    omitted (but including them is not an error).
+                  items:
+                    type: integer
+                    format: int64
+                  example: [4571, 4572]
+      responses:
+        201:
+          description: Message created
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Message"
+        403:
+          description: >
+            Forbidden.  Returned if the user does not have permission to post messages to this room.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  banned:
+                    type: boolean
+                    description: >
+                      True if permission was denied because the user is banned, omitted otherwise.
+                  noWrite:
+                    type: boolean
+                    description: >
+                      True if permission was denied because the user is lacking write permission,
+                      for example because of default room settings or a restriction applied to the
+                      user.
+                  reason:
+                    type: string
+                    description: >
+                      A short, English-language string describing the reason the failure occured.
+                      This is for use as a fallback ONLY: clients should prefer to identify failure
+                      through examining the other fields in the response and only use `reason` as
+                      a user-facing failure description if no known failure is included (which might
+                      happen if the server speaks a new version of the protocol than the client
+                      understands).
+                    example: "Future horrible failure reason"
+
+  /room/{roomToken}/message/{messageId}:
+    get:
+      tags: [Messages]
+      summary: Returns a single posted message by ID.
+      parameters:
+        - $ref: "#/components/parameters/pathRoomToken"
+        - name: messageId
+          in: path
+          description: ID of the message to retrieve.
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Message"
+        403:
+          description: Forbidden
+    put:
```

Comment:
Discussed offline: PUT is slightly preferable because the nature of this is to replace the full record.  Will use PATCH for endpoints that do partial updates.